### PR TITLE
Add UI test documentation

### PR DESCRIPTION
**Pull request by rachelmcr**
This documentation covers the basics of our UI tests:

* What user flows the UI tests cover
* How to run the UI tests locally
* How to add or update UI tests

Update release notes:

- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WooCommerce/WooCommerceUITests/README.md [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+When adding a new UI test, consider:
+
+* Whether you need to test a user flow (to accomplish a task or goal) or a specific feature (e.g. boundary testing).
+* What screens are being tested (defined as screen objects in the [Screens](WooCommerceUITests/Screens) directory).
+* Whether there are actions or flows that could be shared across tests (defined in the [Utils](WooCommerceUITests/Utils) directory).
+* What network requests are made during the test (defined in the [Mocks](WooCommerceUITests/Mocks) directory).
+
+It's preferred to focus UI tests on entire user flows, and group tests with related flows or goals in the same test suite.
+
+When you add a new test, you may need to add new screens and methods. We use screen (page) objects and method chaining for clarity in our tests. Wherever possible, use an existing `accessibilityIdentifier` (or add one to the app) instead of a string to select a UI element on the screen. This ensures tests can be run regardless of the device language.
```
Comment (mokagio) [inline WooCommerce/WooCommerceUITests/README.md:35]: It'd be good to add a reference to what a screen object test is.

I'm not sure there are good ones for mobile (_nudge to write our own_?). I've been pointing people to [Martin Fowler's definition](https://www.martinfowler.com/bliki/PageObject.html) in the past.

Comment (rachelmcr) [reply WooCommerce/WooCommerceUITests/README.md:35]: Good call. When I wrote the WPiOS tests I just called them "page objects" in the docs to use a more commonly understood term, but even then it'd be good to have a reference for clarity.

**Comment by rachelmcr** [comment]
> I wonder if it might be better to link to files at a specific commit, so that future readers won't risk getting 404s or code that is not in the same state as it was when this was written.

I think that's most useful for references to specific lines of code. In this case, where we're referring to files and directories for general reference, I'd lean toward not linking to specific commits because it'll be harder to notice if you're looking at something that's out of date (e.g. old build instructions or CI config) and it'll be hard to remember to update them.

I'd also like to start building the habit of updating our docs when we update the test coverage, since we occasionally get questions about what user flows the UI tests cover. As we expand the number of tests we can revisit how/whether we link to the tests themselves (e.g. just link to `Tests/` and include a plain list of the flows covered by tests).

**Comment by shiki** [comment]
Thank you for documenting, @rachelmcr! 

I would have liked this to be under the [`docs/` documentation tree](https://github.com/woocommerce/woocommerce-ios/tree/develop/docs) so it's a first-class citizen and possibly more discoverable. 🙂

**Comment by rachelmcr** [comment]
Ah, good point @shiki! I have been documenting our UI tests close to the code (something I was used to with other projects) but I'm happy to move it under `docs/`.

That'd be a good opportunity to think about/promote documenting our testing practices more broadly, as well. I'll open a PR soon to move this documentation and maybe add a little more general info.

**Pull request merged**